crates/fluvio-cli/src/consume/mod.rs

```diff
@@ -684,6 +702,12 @@ impl ConsumeOpt {
         } else if let Some(maybe_tail) = self.tail {
             let tail = maybe_tail.unwrap_or(DEFAULT_TAIL);
             Offset::from_end(tail)
+        } else if let Some(maybe_end_offset) = self.end_offset {
```

Comment:
I think that this block is not needed

Reply:
Yeah makes sense. I think we were on the right track initially, based on sehyo's updated comments above. I'll fix this.